data-in-pipeline/app/extract/navigator_family.py

```diff
@@ -1,39 +1,123 @@
+import logging
+from datetime import datetime
+
 import requests
 from pydantic import BaseModel
+from requests.adapters import HTTPAdapter, Retry
+
+from app.connector import NavigatorConnectorConfig
+from app.models import ExtractedEnvelope
+from app.util import generate_envelope_uuid
 
-from app.models import Extracted
-from app.util import get_api_url
+logger = logging.getLogger(__name__)
 
 
 class NavigatorFamily(BaseModel):
     import_id: str
 
 
-API_BASE_URL = get_api_url()
+def extract_navigator_family(
+    import_id: str, config: NavigatorConnectorConfig
+) -> ExtractedEnvelope:
+    """
+    Extract a single family from Navigator API.
+
+    :param import_id: Family identifier
+    :param config: Typed connector configuration
+    :return ExtractedEnvelope with family data and metadata
+
+    Raises:
+        requests.HTTPError: On non-retryable HTTP errors
+        requests.Timeout: If request exceeds timeout
+    """
+
+    logger.info(f"Extracting family {import_id} from Navigator")
+
+    try:
+        family = _fetch_with_retry(import_id, config)
+
+        envelope = ExtractedEnvelope(
+            data=family,
+            envelope_id=generate_envelope_uuid(),
+            source_name="navigator-families",
+            source="navigator",
```

Comment:
Same naming questions as above, can we use `navigator_family` and what is the difference between the two??